src/auth.ts

```diff
@@ -3,32 +3,59 @@ import * as os from 'os';
 import * as path from 'path';
 import * as core from '@actions/core';
 import * as io from '@actions/io';
+import * as exec from '@actions/exec';
 
 export const M2_DIR = '.m2';
 export const SETTINGS_FILE = 'settings.xml';
+export const PRIVATE_KEY_DIR = '.keys';
+export const PRIVATE_KEY_FILE = 'private-key.asc';
 
 export const DEFAULT_ID = 'github';
 export const DEFAULT_USERNAME = 'GITHUB_ACTOR';
 export const DEFAULT_PASSWORD = 'GITHUB_TOKEN';
+export const DEFAULT_GPG_PRIVATE_KEY = '';
+export const DEFAULT_GPG_PASSPHRASE = 'GPG_PASSPHRASE';
```

Comment:
Should we default to an empty string here?

Reply:
This does default to an empty string. Just like GITHUB_TOKEN and GITHUB_ACTOR are environment variables, GPG_PASSPHRASE is an environment variable as well.